analyses/rna-seq-protocol-dge/03-protocol-dge-seg.R

```diff
@@ -0,0 +1,315 @@
+table_outdir <- file.path('results', paste0(norm_method, '_normalized'))
+dir.create(table_outdir, showWarnings = FALSE)
+
+plot_outdir <- file.path('plots', paste0(norm_method, '_normalized'))
+dir.create(plot_outdir, showWarnings = FALSE)
+#------------ Read and pre-process data --------------------
+print('Read count matrices...')
+polya <- readRDS(
+    "results/pbta-gene-counts-rsem-expected_count-collapsed.polya.rds")
+stranded <- readRDS(
+    "results/pbta-gene-counts-rsem-expected_count-collapsed.stranded.rds")
+
+common_gene_id <- intersect(rownames(stranded), rownames(polya))
+# 6 samples have both poly-A and stranded RNA-seq libraries
+# Index | Kids_First_Biospecimen_ID | sample_id | experimental_strategy | 
+#   RNA_library | cohort
+# -- | -- | -- | -- | -- | --
+# 1 | BS_HE0WJRW6 | 7316-1455 | RNA-Seq | stranded | CBTN
+# 2 | BS_HWGWYCY7 | 7316-1455 | RNA-Seq | poly-A | CBTN 
+# 3 | BS_SHJA4MR0 | 7316-161 | RNA-Seq | stranded | CBTN 
+# 4 | BS_X0XXN9BK | 7316-161 | RNA-Seq | poly-A | CBTN 
+# 5 | BS_FN07P04C | 7316-255 | RNA-Seq | stranded | CBTN 
+# 6 | BS_W4H1D4Y6 | 7316-255 | RNA-Seq | poly-A | CBTN
+# 7 | BS_8QB4S4VA | 7316-536 | RNA-Seq | stranded | CBTN
+# 8 | BS_QKT3TJVK | 7316-536 | RNA-Seq | poly-A | CBTN
+# 9 | BS_7WM3MNZ0 | A16915 | RNA-Seq | poly-A | PNOC003
+# 10 | BS_KABQQA0T | A16915 | RNA-Seq | stranded | PNOC003
+# 11 | BS_68KX6A42 | A18777 | RNA-Seq | poly-A | PNOC003
+# 12 | BS_D7XRFE0R | A18777 | RNA-Seq | stranded | PNOC003
+counts <- cbind(
+    stranded[common_gene_id, c("BS_HE0WJRW6", "BS_SHJA4MR0", "BS_FN07P04C",
+                               "BS_8QB4S4VA", "BS_KABQQA0T", "BS_D7XRFE0R")],
+    polya[common_gene_id, c("BS_HWGWYCY7", "BS_X0XXN9BK", "BS_W4H1D4Y6",
+                            "BS_QKT3TJVK", "BS_7WM3MNZ0", "BS_68KX6A42")]
+)
+group <- factor(c(rep("stranded", 6), rep("polya", 6)))
+# Make design matrix 
+design <- model.matrix(~group)
+
+# sample_prot is used to select stably expressed genes
+sample_prot <- c('s1_stranded', 's2_stranded', 's3_stranded',
+                 's4_stranded', 's5_stranded', 's6_stranded',
+                 's1_polya', 's2_polya', 's3_polya',
+                 's4_polya', 's5_polya', 's6_polya')
+
+### build normalized DGEList
+counts_object <- DGEList(counts = counts, group = group)
+counts_object <- counts_object[
+    filterByExpr(counts_object), , keep.lib.sizes=FALSE]
+
+if (identical(norm_method, 'tmm')) {
+    counts_object <- calcNormFactors(counts_object, method="TMM")
+    norm_cnt_mat <- cpm(counts_object)
+    round_norm_cnt_mat <- round(norm_cnt_mat)
+    plot_nm_str <- 'TMM'
+} else if (identical(norm_method, 'uqpgq2')) {
+    counts_object <- DGEList(counts = uq.pgQ2(X = counts_object$counts),
+                            group = group)
+    norm_cnt_mat <- counts_object$counts
+    round_norm_cnt_mat <- round(norm_cnt_mat)
+    counts_object <- calcNormFactors(counts_object, method="none")
+    plot_nm_str <- 'UQ-pgQ2'
+}
+### Estimate dispersion
```

Comment:
Seems the only step missing for all DGE analyses (edger and deseq2) in this module is subsetting down to housekeeping genes (per this ticket comment (https://github.com/PediatricOpenTargets/ticket-tracker/issues/17#issue-896875722)). 
So there would only be one modification to the current steps (step 3): 
1. Filter out non-expressed genes on the full transcriptome
2. Correct for compositional biases on the full transcriptome (TMM, uqpgq2, or DESeq2's method). 
3. _Filter down to housekeeping genes from https://housekeeping.unicamp.br/?download (Full Housekeeping gene list, ~2000 genes)_
4. Estimate dispersions
5. Run DGE and plot p-val histograms.